Source/HTTP/Android/AndroidHttpProvider.cpp

```diff
@@ -0,0 +1,85 @@
+#include "pch.h"
+#include <httpClient/httpClient.h>
+#include <httpClient/httpProvider.h>
+#include "AndroidHttpProvider.h"
+#include "android_http_request.h"
+#include "Platform/Android/PlatformComponents_Android.h"
+
+NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
+
+AndroidHttpProvider::AndroidHttpProvider(SharedPtr<PlatformComponents_Android> platformComponents) :
+    m_platformComponents{ std::move(platformComponents) }
+{
+}
+
+HRESULT AndroidHttpProvider::PerformAsync(
+    HCCallHandle call,
+    XAsyncBlock* asyncBlock
+) noexcept
+{
+    std::unique_ptr<HttpRequest> httpRequest{
+        new HttpRequest(
+            asyncBlock,
+            m_platformComponents->GetJavaVm(),
+            m_platformComponents->GetApplicationContext(),
+            m_platformComponents->GetHttpRequestClass(),
+            m_platformComponents->GetHttpResponseClass()
+        )
+    };
+
+    HRESULT result = httpRequest->Initialize();
+
+    if (!SUCCEEDED(result))
+    {
+        HCHttpCallResponseSetNetworkErrorCode(call, result, 0);
+        XAsyncComplete(asyncBlock, result, 0);
```

Comment:
better to do:
        HCHttpCallResponseSetNetworkErrorCode(call, result, result);

cause 0 = S_OK

Reply:
changed